accounting/src/delta/delta_data_collection/tests/delta_delta_delta_delta_tests.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+use DataDelta::{Create, Delete, Modify};
+
+use rstest::rstest;
+
+type FourCollections = (
+    DeltaDataCollection<i32, char>,
+    DeltaDataCollection<i32, char>,
+    DeltaDataCollection<i32, char>,
+    DeltaDataCollection<i32, char>,
+);
+
+fn make_expected_collection(delta: DataDelta<char>) -> DeltaDataCollection<i32, char> {
+    DeltaDataCollection::from_iter([(1, delta)])
+}
+
+fn make_four_collections(
+    delta1: DataDelta<char>,
+    delta2: DataDelta<char>,
+    delta3: DataDelta<char>,
+    delta4: DataDelta<char>,
+) -> FourCollections {
+    (
+        DeltaDataCollection::from_iter([(1, delta1)]),
+        DeltaDataCollection::from_iter([(1, delta2)]),
+        DeltaDataCollection::from_iter([(1, delta3)]),
+        DeltaDataCollection::from_iter([(1, delta4)]),
+    )
+}
+
+#[rstest]
+#[case(
```

Comment:
I don't think these cases cover all the possibilities. I'm even OK with writing a little script to generate everything possible.